src/main/java/de/tum/in/www1/artemis/web/rest/PlagiarismResource.java

```diff
@@ -1,46 +1,238 @@
     @PreAuthorize("hasRole('EDITOR')")
-    public ResponseEntity<Void> updatePlagiarismComparisonStatus(@PathVariable long comparisonId, @RequestBody PlagiarismComparisonStatusDTO statusDTO) {
-        // TODO: check that the editor has access to the corresponding course (add the exerciseId to the URL)
+    public ResponseEntity<Void> updatePlagiarismComparisonStatus(@PathVariable("courseId") long courseId, @PathVariable("comparisonId") long comparisonId,
+            @RequestBody PlagiarismComparisonStatusDTO statusDTO) {
         log.debug("REST request to update the status of the plagiarism comparison with id: {}", comparisonId);
+        Course course = courseRepository.findByIdElseThrow(courseId);
+        User currentUser = userRepository.getUserWithGroupsAndAuthorities();
+        if (!authenticationCheckService.isAtLeastEditorInCourse(course, currentUser)) {
+            throw new AccessForbiddenException("Only editors or instructors of this course can update plagiarism cases.");
+        }
         var comparison = plagiarismComparisonRepository.findByIdElseThrow(comparisonId);
         plagiarismComparisonRepository.updatePlagiarismComparisonStatus(comparison.getId(), statusDTO.getStatus());
         return ResponseEntity.ok().body(null);
     }
+
+    /**
+     * Retrieves all plagiarismCases related to a course that were previously confirmed.
+     *
+     * @param courseId the id of the course
+     * @return all plagiarism cases
+     */
+    @GetMapping("courses/{courseId}/plagiarism-cases")
+    @PreAuthorize("hasRole('INSTRUCTOR')")
+    public ResponseEntity<List<PlagiarismCaseDTO>> getPlagiarismCasesForCourse(@PathVariable long courseId) {
+        log.debug("REST request to get all plagiarism cases in course with id: {}", courseId);
+        Course course = courseRepository.findByIdElseThrow(courseId);
+        if (!authenticationCheckService.isAtLeastInstructorInCourse(course, userRepository.getUserWithGroupsAndAuthorities())) {
+            throw new AccessForbiddenException("Only instructors of this course have access to its plagiarism cases.");
+        }
+        ArrayList<PlagiarismCaseDTO> foundPlagiarismCasesForCourse = this.plagiarismService.collectAllPlagiarismCasesForCourse(courseId);
+        return ResponseEntity.ok(foundPlagiarismCasesForCourse);
+    }
+
+    /**
+     * Updates an instructor statement on a plagiarismComparison (for one side).
+     * This process will send a notification to the respective student.
+     * I.e. the instructor sets a personal message to one of the accused students.
+     *
+     * @param comparisonId the id of the PlagiarismComparison
+     * @param studentLogin of one of accused students
+     * @param statement of the instructor directed to one of the accused students
+     * @return the instructor statement (convention)
+     */
+    @PutMapping("plagiarism-comparisons/{comparisonId}/{studentLogin}/instructor-statement")
```

Comment:
does not follow our REST URL best practices, please improve it:
- start with courses/{courseId}
- studentLogin should not follow immediately after the id, instead you need something in between, such as /cases/